src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo.java

```diff
@@ -190,6 +189,7 @@ public void testTreeJsonCircularDependency() throws IOException {
     public void testTreeJsonParsing() throws Exception {
         List<String> contents = runTreeMojo("tree2.json", "json");
 
+        System.setProperty("jakarta.json.provider", "org.glassfish.json.JsonProviderImpl");
```

Comment:
Why is this necessary? Is there not `ServiceProvider` file in `META-INF?

Reply:
Right now there doesn't seem to be. I can add it if you prefer but that will clutter the main jar with details only needed for the test.